libvast/src/system/type_registry.cpp

```diff
@@ -102,30 +113,56 @@ type_registry(type_registry_actor self, const path& dir) {
         self, "failed to persist state to disk:", self->system().render(err));
     self->quit(msg.reason);
   });
-  return {[=](telemetry_atom) {
-            // Send out a heartbeat.
-            self->send(self->state.accountant, self->state.telemetry());
-            self->delayed_send(self, defaults::system::telemetry_rate,
-                               telemetry_atom::value);
-          },
-          [=](status_atom) {
-            // Send out a status report.
-            return self->state.status();
-          },
-          [=](caf::stream<table_slice_ptr> in) {
-            // Store layout of every incoming table slice.
-            caf::attach_stream_sink(
-              self, in,
-              [=](caf::unit_t&) { VAST_DEBUG(self, "initialized stream"); },
-              [=](caf::unit_t&, table_slice_ptr x) {
-                VAST_TRACE(self, "received new table slice");
-                self->state.insert(std::move(x->layout()));
-              });
-          },
-          [=](std::string name) {
-            // Retrieve a list of known states for a name.
-            return self->state.lookup(name);
-          }};
+  // Load existing state from disk if possible.
+  if (auto err = self->state.load_from_disk())
+    self->quit(std::move(err));
+  // Load already known types from the event_types singleton.
+  auto schema = vast::event_types::get();
+  if (!schema)
+    VAST_ERROR(self, "was unable to retrieve global event types");
```

Comment:
I think the user is still left wondering what happened with this error message. How about relating it to the schemas on disk, which apparently failed to load?

Reply:
I've made it an assertion. This code only runs during startup and should never fail.